src/WtsApi32.Desktop/WtsApi32+WtsSafeMemoryGuard.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) to owners found in https://github.com/AArnott/pinvoke/blob/master/COPYRIGHT.md. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+
+namespace PInvoke
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Runtime.InteropServices;
+    using System.Text;
+
+    /// <content>
+    /// Exported functions from the WtsApi32.dll Windows library
+    /// that are available to Desktop apps only.
+    /// </content>
+    public static partial class WtsApi32
+    {
+        /// <summary>
+        /// Base guard for all memory allocated inside WtsApi32.
+        /// </summary>
+        public partial class WtsSafeMemoryGuard : SafeHandle
```

Comment:
It has Read<T> Write<T> but their both take byteOffset argument, that not so different as IntPtr. As addition I see only bounds checking, but I still have them. As base class for SafeWtsMemory it is good, but for what I'm trying to achieve with the iterator, it is not so much replace.
Though, I'll use it. Thanks, didn't even knew about it.